includes/reader-activation/class-reader-activation.php

```diff
@@ -2063,41 +2059,5 @@ public static function rate_limit_lost_password( $errors, $user_data ) {
 		}
 		return $errors;
 	}
-
-	/**
-	 * Automatically force deactivate the ESP sync on staging sites to prevent polluting the data in ESPs.
-	 *
-	 * @param mixed  $value Setting value.
-	 * @param string $setting Setting name.
-	 * @return mixed Possibly modified $value.
-	 */
-	public static function disable_esp_sync_on_staging_sites( $value, $setting ) {
-		if ( 'sync_esp' !== $setting ) {
-			return $value;
-		}
-
-		if ( defined( 'NEWSPACK_FORCE_ALLOW_ESP_SYNC' ) && NEWSPACK_FORCE_ALLOW_ESP_SYNC ) {
-			return $value;
-		}
-
-		$site_url = strtolower( \untrailingslashit( \get_site_url() ) );
-		if ( false !== stripos( $site_url, '.newspackstaging.com' ) ) {
-			return false;
-		}
-
-		// Neither WCS_Staging::is_duplicate_site() nor is_plugin_active() are initialized early enough for all situations.
-		// So we need to re-create the logic from both.
-		if ( in_array( 'woocommerce-subscriptions/woocommerce-subscriptions.php', (array) \get_option( 'active_plugins', [] ), true ) ) {
-			$subscriptions_site_url = \get_option( 'wc_subscriptions_siteurl', false );
-			if ( $subscriptions_site_url ) {
-				$cleaned_subscriptions_site_url = strtolower( untrailingslashit( str_ireplace( '_[wc_subscriptions_siteurl]_', '', $subscriptions_site_url ) ) );
-				if ( $cleaned_subscriptions_site_url !== $site_url ) {
-					return false;
-				}
-			}
-		}
-
-		return $value;
-	}
 }
```

Comment:
Removing this as we should always rely on `Newspack\Reader_Activation\ESP_Sync::can_esp_sync()`.